pkg/pipeline/service.go

```diff
@@ -102,6 +112,15 @@ func (s *Service) Get(ctx context.Context, id string) (*Instance, error) {
 // Create will create a new pipeline instance with the given config and return
 // it if it was successfully saved to the database.
 func (s *Service) Create(ctx context.Context, id string, cfg Config, p ProvisionType) (*Instance, error) {
+	pl, err := s.createInternal(ctx, id, cfg, p)
+	if err != nil {
+		s.notify(id, p, err)
```

Comment:
I don't think we should notify on creation, this error will be returned by the provisioning service.

Reply:
This is to cover the API-created pipelines as well.

Reply:
If we _really_ want to do that we should also make sure that we catch errors in the connectors and processors service. That said, I honestly think it's going too far if we do this, if I fail to _create_ a pipeline, connector or processor for some reason I really don't think Conduit should crash 😕 If I manage to start the pipeline and it fails because of an error that's one thing, but not if I make a mistake while creating the pipeline.

Reply:
I'd say this is taking it a step to far and is a pretty bad UX. I'd say we should stop if something fails to create, since that doesn't really address the _spirit_ of the initial request.

Reply:
+1 on validate errors in creation via the API shouldn't just kill the process.

Reply:
It's no problem to change, and I don't feel strongly about this. I do agree that this is pretty "brutal". 

However, WDYT about the same for pipeline config files? Using this example: because of the pipeline name, one of the pipelines won't be provided (current behavior in the main branch). That, together with the `exit-on-error` flag, makes Conduit crash.

Personally, I think the behavior as it is in this PR, is consistent (though very strict). I'm also assuming that this flag will typically be used in production environments where pipeline config files will mostly be used.

Reply:
I think these two ways to create the pipeline are fundamentally different in the sense of the feedback you get. The API gives you immediate feedback about the success of the actions you are performing, while provisioning the pipeline through a config file is "asynchronous", you need to check the logs or the API to see what happened. And as far as I understand, this issue should be addressing the second part + any errors that happen if a running pipeline fails.

Reply:
The feedback perspective you're mentioning is a good one and I agree with that and is a good argument against being consistent in all aspects.:)

With everything mentioned above, I'll change this **not** to fail for API pipelines.